docs/developer/developer-handbook.md

```diff
@@ -0,0 +1,93 @@
+# EDC Developer Handbook
+
+<!-- TOC -->
+* [EDC Developer Handbook](#edc-developer-handbook)
+  * [Introduction](#introduction)
+  * [Building a distribution](#building-a-distribution)
+    * [Seed some data](#seed-some-data)
+    * [Perform a data transfer](#perform-a-data-transfer)
+  * [The control plane - core concepts](#the-control-plane---core-concepts)
```

Comment:
I would split core concepts from the control plane

Reply:
but it specifically talks about core concepts _of_ the control plane

Reply:
some those core concepts are in fact valid for all the components, not only the connector, like state machines, extension model and so on

Reply:
i don't see much use in explaining generic state machines, we can assume people know what that is. 
I would rather outline the specific ones we have, i.e. contract negotiation and transfer process state machine, which are of the control plane. 
We could list some general principles, like asynchronicity though.